tests/components/accuweather/test_weather.py

```diff
@@ -0,0 +1,148 @@
+    await init_integration(hass, forecast=True)
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    state = hass.states.get("weather.home")
+    assert state
+    assert state.state == "sunny"
+    assert state.attributes.get(ATTR_WEATHER_HUMIDITY) == 67
+    assert state.attributes.get(ATTR_WEATHER_OZONE) == 32
+    assert state.attributes.get(ATTR_WEATHER_PRESSURE) == 1012.0
+    assert state.attributes.get(ATTR_WEATHER_TEMPERATURE) == 22.6
+    assert state.attributes.get(ATTR_WEATHER_VISIBILITY) == 16.1
+    assert state.attributes.get(ATTR_WEATHER_WIND_BEARING) == 180
+    assert state.attributes.get(ATTR_WEATHER_WIND_SPEED) == 14.5
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    forecast = state.attributes.get(ATTR_FORECAST)[0]
+    assert forecast.get(ATTR_FORECAST_CONDITION) == "lightning-rainy"
+    assert forecast.get(ATTR_FORECAST_PRECIPITATION) == 4.8
+    assert forecast.get(ATTR_FORECAST_PRECIPITATION_PROBABILITY) == 58
+    assert forecast.get(ATTR_FORECAST_TEMP) == 29.5
+    assert forecast.get(ATTR_FORECAST_TEMP_LOW) == 15.4
+    assert forecast.get(ATTR_FORECAST_TIME) == "2020-07-26T05:00:00+00:00"
+    assert forecast.get(ATTR_FORECAST_WIND_BEARING) == 166
+    assert forecast.get(ATTR_FORECAST_WIND_SPEED) == 13.0
+
+    entry = registry.async_get("weather.home")
+    assert entry
+    assert entry.unique_id == "0123456"
+
+
+async def test_availability(hass):
+    """Ensure that we mark the entities unavailable correctly when service is offline."""
+    await init_integration(hass)
+
+    state = hass.states.get("weather.home")
+    assert state
+    assert state.state != STATE_UNAVAILABLE
+    assert state.state == "sunny"
+
+    future = utcnow() + timedelta(minutes=60)
+    with patch(
+        "accuweather.AccuWeather._async_get_data", side_effect=ConnectionError()
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        state = hass.states.get("weather.home")
+        assert state
+        assert state.state == STATE_UNAVAILABLE
+
+    future = utcnow() + timedelta(minutes=120)
+    with patch(
+        "accuweather.AccuWeather.async_get_current_conditions",
+        return_value=json.loads(
+            load_fixture("accuweather/current_conditions_data.json")
+        ),
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        state = hass.states.get("weather.home")
+        assert state
+        assert state.state != STATE_UNAVAILABLE
+        assert state.state == "sunny"
+
+
+async def test_manual_update_entity(hass):
+    """Test manual update entity via service homeasasistant/update_entity."""
+    await init_integration(hass, forecast=True)
+
+    await async_setup_component(hass, "homeassistant", {})
+    with patch(
+        "accuweather.AccuWeather.async_get_current_conditions"
+    ) as mock_current, patch(
```

Comment:
These patches don't have return data.  Is it being returned an AsyncMock object that is causing the issue?

Reply:
ah valid point, ill try that next

Reply:
@MatthewFlamm You were right, now the tests passed without a problem.